EfYouTests/Filters/FilterServiceOfTTests.cs

```diff
@@ -68,24 +67,43 @@ public void FilterResultsOnGet_PrimaryKeyOfTypeIsLong_ReturnsQueryableFilteredBy
             var queryable = new List<DummyParent> { new DummyParent { Id = 1 }, new DummyParent { Id = 2 }, new DummyParent { Id = 3 } }.AsQueryable();
 
             // Act
-            var results = filterService.Object.FilterResultsOnGet(queryable, new List<long> { 2 });
+            var results = filterService.Object.FilterResultsOnGet(queryable, new List<dynamic> { 2 });
 
             // Assert
             Assert.AreEqual(1, results.Count()); // Only one result should pass the above defined filter on filter on ids.
             Assert.AreEqual(2, results.Single().Id); // The single result should have Id = 2.
         }
 
+        [TestMethod]
+        public void FilterResultsOnGet_PrimaryKeyOfTypeIsGuid_ReturnsQueryableFilteredByGuidIds()
+        {
+            // Arrange
+            var filterService = new Mock<FilterService<DummyEntityWithGuidId>> { CallBase = true };
+            var first = Guid.NewGuid();
+            var second = Guid.NewGuid();
+            var third = Guid.NewGuid();
+
+            var queryable = new List<DummyEntityWithGuidId> { new DummyEntityWithGuidId { Id = first }, new DummyEntityWithGuidId { Id = second }, new DummyEntityWithGuidId { Id = third } }.AsQueryable();
+
+            // Act
+            var results = filterService.Object.FilterResultsOnGet(queryable, new List<dynamic> { second });
+
+            // Assert
+            Assert.AreEqual(1, results.Count()); // Only one result should pass the above defined filter on filter on ids.
+            Assert.AreEqual(second, results.Single().Id); // The single result should have Id = second GUID.
```

Comment:
Don't need comments here, you're code already reads pretty well for that.

If you feel you really need comments though explaining this, maybe chuck what should be happening in a big problem statement at the top of the file? It seems you explain this more than once so it might be worth a shot.

Reply:
Probably true that comments aren't needed.  I'll remove them.